hibernate-core/src/main/java/org/hibernate/id/enhanced/SequenceStyleGenerator.java

```diff
@@ -521,4 +572,39 @@ public String determineBulkInsertionIdentifierGenerationSelectFragment(Dialect d
 	public void registerExportables(Database database) {
 		databaseStructure.registerExportables( database );
 	}
+
+	protected Integer determineDatabaseIncrementSize(ServiceRegistry serviceRegistry, String sequenceName) {
+		JdbcConnectionAccess connectionAccess = serviceRegistry.getService( JdbcServices.class )
+				.getBootstrapJdbcConnectionAccess();
+		Connection connection = null;
+		try {
+			connection = connectionAccess.obtainConnection();
+
+			try (PreparedStatement statement = connection.prepareStatement(
+					"SELECT increment " +
+							"FROM information_schema.sequences " +
+							"WHERE sequence_name = ?" )) {
```

Comment:
I'll have to check to see how to accommodate `sequence_catalog` and `sequence_schema` so that it works on all major DBs.

Reply:
Some databases do in fact allow having same named sequences in different namespaces

Reply:
I tested it on Oracle, SQL Server, PostgreSQL and H2 and it worked on all of them as the Information_Schema should, theoretically, be defined by the SQL Standard. However, it could be that some DBs will not honor the same structure, in which case, the request will not break since we don't escalate the exception up the stack. It would just be that Hibernate cannot determine the increment size. Nevertheless, I could push this query into the base Dialect, so we could override it for those DBs which don't follow the SQL Standard guidelines for Information Schema.